### PR TITLE
[CoreFoundation] [tests] Add new DispatchQueue overloaded method

### LINKED ISSUE #6783: Add `DispatchQueue.GetGlobalQueue (DispatchQualityOfService)` overloaded method
### Steps to Reproduce

1. Type `DispatchQueue.GetGlobalQueue (` in a .cs file

### Expected Behavior

See an overloaded method supporting `DispatchQualityOfService` enum.

### Actual Behavior

No overloaded method supporting `DispatchQualityOfService` enum.

### More info

According to [Apple documentation][1], now `dispatch_get_global_queue` (a.k.a `DispatchQueue.GetGlobalQueue`) should support `DispatchQualityOfService` enum. This is not a blocking issue due to according to `queue.h` header, if you pass a `DispatchQueuePriority`, the method will map it with a `DispatchQualityOfService`:

```obj-c
/*
 * The global concurrent queues may still be identified by their priority,
 * which map to the following QOS classes:
 *  - DISPATCH_QUEUE_PRIORITY_HIGH:         QOS_CLASS_USER_INITIATED
 *  - DISPATCH_QUEUE_PRIORITY_DEFAULT:      QOS_CLASS_DEFAULT
 *  - DISPATCH_QUEUE_PRIORITY_LOW:          QOS_CLASS_UTILITY
 *  - DISPATCH_QUEUE_PRIORITY_BACKGROUND:   QOS_CLASS_BACKGROUND
 */
```

But it would be nice to have an overloaded method that supports `DispatchQualityOfService` enum.

### Environment

```
=== Visual Studio Enterprise 2019 for Mac ===

Version 8.2.2 (build 44)
Installation UUID: 48b5fb7e-9587-45b0-b10e-58dc44b4c3fe
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.14.0.85 (d16-2-fix-mac-updates / 86973979)

	Package version: 600000319

=== Mono Framework MDK ===

Runtime:
	Mono 6.0.0.319 (2019-02/52203786470) (64-bit)
	Package version: 600000319

=== NuGet ===

Version: 5.1.0.6013

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.0.0-preview3-27503-5
	2.2.3
	2.1.9
	2.1.8
	2.1.7
	2.1.2
	2.1.1
	2.0.5
SDK: /usr/local/share/dotnet/sdk/2.2.105/Sdks
SDK Versions:
	2.2.105
	2.1.505
	2.1.504
	2.1.503
	2.1.302
	2.1.301
	2.1.4
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.0.0/lib/mono/msbuild/Current/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.11
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 9.4.1.0 (Visual Studio Enterprise)
Commit: xamarin-android/d16-2/ebd65c9
Android SDK: /Users/israelsoto/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.4 (API level 19)
		6.0 (API level 23)
		7.0 (API level 24)
		7.1 (API level 25)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.2
SDK Build Tools Version: 29.0.0 rc3

Build Information: 
Mono: mono/mono/2019-02@e6f5369c2d2
Java.Interop: xamarin/java.interop/d16-2@d64ada5
LibZipSharp: grendello/LibZipSharp/d16-2@caa0c74
LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-2@6f6c969

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/israelsoto/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android Device Manager ===

Version: 1.2.0.58
Hash: d03e3ea
Branch: remotes/origin/d16-2~1
Build date: 2019-08-05 20:08:02 UTC

=== Apple Developer Tools ===

Xcode 11.0 (14908.4)
Build 11M382q

=== Xamarin.Mac ===

Version: 5.99.2.40 (Visual Studio Enterprise)
Hash: d883febf
Branch: xcode11
Build date: 2019-08-06 08:33:41-0400

=== Xamarin.iOS ===

Version: 12.99.2.40 (Visual Studio Enterprise)
Hash: d883febf
Branch: xcode11
Build date: 2019-08-06 08:33:40-0400

=== Xamarin Designer ===

Version: 16.2.0.401
Hash: a8111869d
Branch: remotes/origin/d16-2-xcode11
Build date: 2019-08-05 20:44:10 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 802020044
Git revision: 2be0fd83c12d2bf1d60b2e5413d8517b8779bb74
Build date: 2019-08-07 07:47:32+00
Build branch: release-8.2-xcode11
Xamarin extensions: 559d23bf0f48ccb83b732417a122fc66e0a2640a

=== Operating System ===

Mac OS X 10.14.6
Darwin 18.7.0 Darwin Kernel Version 18.7.0
    Thu Jun 20 18:42:21 PDT 2019
    root:xnu-4903.270.47~4/RELEASE_X86_64 x86_64
```

[1]: https://developer.apple.com/documentation/dispatch/1452927-dispatch_get_global_queue

### PR DESCRIPTION
**Pull request by michizhou**
Add new `DispatchQueue.GetGlobalQueue` overloaded method that is fully capable of supporting `DispatchQualityOfService` enum in `Dispatch.cs`. Also added appropriate unit tests within `DispatchTests.cs` and other associated test scripts that call `DispatchQueue.GetGlobalQueue`.

Fixes #6783

### CODE REVIEW

**Comment by spouliot** [comment]
build

**Comment by dalexsoto** [comment]
build

File: src/CoreFoundation/Dispatch.cs [hunk 1]
```diff
@@ -530,7 +535,11 @@ public DispatchQualityOfService GetQualityOfService (out int relative_priority)
 
 		[DllImport (Constants.libcLibrary)]
 		// dispatch_queue_t dispatch_get_global_queue (long priority, unsigned long flags);
-		extern static IntPtr dispatch_get_global_queue (nint priority, nuint flags);
+		extern static IntPtr dispatch_get_global_queue_priority (nint priority, nuint flags);
```
Comment (rolfbjarne) [inline src/CoreFoundation/Dispatch.cs]: As far as I can tell there's no function called `dispatch_get_global_queue_priority` in any of the platforms we support.

File: tests/bindings-test/ApiDefinition.cs [hunk 2]
```diff
@@ -376,14 +376,21 @@ interface ObjCBlockTester
 		void CallOptionalStaticCallback ();
 
 		[Static]
-		[Export ("callAssertMainThreadBlockRelease:")]
```
Comment (rolfbjarne) [inline tests/bindings-test/ApiDefinition.cs]: You're removing an existing method: that's a breaking change, so we can't do that.

File: tests/bindings-test/ApiDefinition.cs [hunk 3]
```diff
@@ -376,14 +376,21 @@ interface ObjCBlockTester
...
+		void CallAssertMainThreadBlockReleasePriority (OuterBlock completionHandler);
+
+		[Static]
+		[Export ("callAssertMainThreadBlockReleaseQOS:")]
+		void CallAssertMainThreadBlockReleaseQOS (OuterBlock completionHandler);
 
 		[Export ("assertMainThreadBlockReleaseCallback:")]
 		void AssertMainThreadBlockReleaseCallback (InnerBlock completionHandler);
 
-		[Export ("callAssertMainThreadBlockReleaseCallback")]
```
Comment (rolfbjarne) [inline tests/bindings-test/ApiDefinition.cs]: Same: you're removing an existing method.

File: tests/bindings-test/RuntimeTest.cs [hunk 4]
```diff
@@ -57,19 +57,36 @@ public void EvilDeallocatorTest ()
...
 		[Test]
-		public void MainThreadDeallocationTest ()
+		public void MainThreadDeallocationTestPriority ()
 		{
 #if OPTIMIZEALL
 			if (!TestRuntime.IsLinkAll)
 				Assert.Ignore ("This test must be processed by the linker if all optimizations are turned on.");
 #endif
 
-			ObjCBlockTester.CallAssertMainThreadBlockRelease ((callback) => {
```
Comment (rolfbjarne) [inline tests/bindings-test/RuntimeTest.cs]: Also keep the existing test code (it should continue to work), while just adding new tests instead.

**Comment by michizhou** [comment]
@rolfbjarne I just changed the code such that the old existing tests have been restored. Please feel free to rerun the build on my commit and let me know about any further changes that are needed.

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
Test failures are unrelated: 

* link all/Mac Modern/Debug: https://github.com/xamarin/maccore/issues/2056
* link all/Mac Modern/Release: https://github.com/xamarin/maccore/issues/2056
* link sdk/Mac Modern/Debug: https://github.com/xamarin/maccore/issues/2056
* link sdk/Mac Modern/Release: https://github.com/xamarin/maccore/issues/2056
* monotouch-test/watchOS 32-bits - simulator/Debug: https://github.com/xamarin/maccore/issues/581
* framework-test/watchOS 32-bits - simulator/Debug: https://github.com/xamarin/maccore/issues/581

**Comment by rolfbjarne** [comment]
Thanks a lot for your contribution!

**Pull request merged**